cypress/integration/moderation/ban_user.spec.js

```diff
@@ -0,0 +1,33 @@
+/// <reference types="Cypress" />
+
+describe('Moderators', function() {
+  beforeEach(function() {
+    cy.task('dropAndSeedDatabase');
+    cy.fixture('users/testAdmin').as('testAdmin');
+    cy.fixture('users/testUser').as('testUser');
+    cy.fixture('posts/testPost').as('bannedUserPost');
+  });
+
+  it('can ban users and remove their content from public view', function() {
+    cy.loginAs(this.testAdmin);
+    cy.visit('/');
+    cy.contains(this.bannedUserPost.title).should('be.visible');
+    cy.visit(`/users/${this.testUser.slug}/edit`);
+    cy.get('.form-section-ban-and-purge-user .form-section-heading-toggle').click();
+    cy.get('.input-deleteContent .MuiCheckbox-root').click()
+    cy.get('input[name="banned"]').click();
+    cy.get('.rdtNext').click(); // jumping to next month via the date picker
+    cy.get('.rdtDay.rdtNew[data-value="1"]').click();
```

Comment:
Thanks for that resource, it looks great! Yeah I've felt reluctant to bind to classes as much as I have, but MaterialUI and (especially) CKEditor do make it complicated and at times untenable to do otherwise with the code as-is. I should make a pass of updating test-related components to have labels/names/etc to foster more principled querying. My sense is that the correct prioritization is for me to not block on this right now, but I've added an Asana task for it.